packages/mui-material/src/Modal/Modal.js

```diff
@@ -101,18 +102,14 @@ const Modal = React.forwardRef(function Modal(inProps, ref) {
   return (
     <ModalUnstyled
       components={{
-        Root: ModalRoot,
+        Root: component ?? ModalRoot,
+        Backdrop: BackdropComponent,
         ...components,
       }}
       componentsProps={{
-        root: {
-          ...componentsProps.root,
-          ...((!components.Root || !isHostComponent(components.Root)) && {
-            ownerState: { ...componentsProps.root?.ownerState },
-          }),
-        },
+        root: componentsProps.root,
```

Comment:
Maybe add the logic for the components & `as` prop here?

Reply:
👍 Good point. I added it here.